pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1970,6 +1934,28 @@ var _ = Describe("controlplanemachinesetgenerator controller on Nutanix", func()
 	var cpms *machinev1.ControlPlaneMachineSet
 	var machine0, machine1, machine2 *machinev1beta1.Machine
 	var machineSet0, machineSet1, machineSet2, machineSet3, machineSet4 *machinev1beta1.MachineSet
+	var withNutanixFailureDomains bool
+
+	createInfrastructure := func() *configv1.Infrastructure {
+		By("Setting up a new infrastructure for the test")
+
+		infra := util.CreateNewInfrastructureObject4NutanixTest(withNutanixFailureDomains)
+		Expect(infra != nil && infra.Spec.PlatformSpec.Nutanix != nil).To(BeTrue())
```

Comment:
True and false evaluations are hard to debug when they fail. You'd be better doing
```suggestion
		Expect(infra).ToNot(BeNil(), "Expected infrastructure object to not be nil")
		Expect(infra.Spec.PlatformSpec.Nutanix).ToNot(BeNil(), "Expected Nutanix platform spec to be populated")
```

Reply:
Fixed